airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/cdc/MongoDbCdcInitializer.java

```diff
@@ -183,9 +213,60 @@ public List<AutoCloseableIterator<AirbyteMessage>> createCdcIterators(
             AirbyteStreamStatusTraceMessage.AirbyteStreamStatus.COMPLETE)))
         .toList();
 
-    return Stream.of(Collections.singletonList(initialSnapshotIterator), cdcStreamsStartStatusEmitters,
-        Collections.singletonList(AutoCloseableIterators.lazyIterator(incrementalIteratorSupplier, null)),
-        cdcStreamsCompleteStatusEmitters).flatMap(Collection::stream).toList();
+    if (startedCdcStreamList.isEmpty()) {
+      LOGGER.info("First sync - no cdc streams have been completed or started");
+      /*
+       * This is the first run case - no initial loads have been started. In this case, we want to run the
+       * iterators in the following order: 1. Run the initial load iterators. This step will timeout and
+       * throw a transient error if run for too long (> 8hrs by default). 2. Run the debezium iterators
+       * with ALL of the incremental streams configured. This is because if step 1 completes, the initial
+       * load can be considered finished.
+       */
+      return Collections.singletonList(
+          AutoCloseableIterators.concatWithEagerClose(
+              Stream.of(
+                  cdcStreamsStartStatusEmitters,
+                  Collections.singletonList(initialSnapshotIterator),
+                  Collections.singletonList(AutoCloseableIterators.lazyIterator(incrementalIteratorSupplier, null)),
```

Comment:
For the debezium iterators, we want to make sure they are only scoped to the streams that have been completed/started. 
Currently, our debezium iterator logic seems to be starting debezium for all streams : 

```
    final var propertiesManager =
        new MongoDbDebeziumPropertiesManager(defaultDebeziumProperties, config.getDatabaseConfig(), incrementalOnlyStreamsCatalog, cdcStreamList);
    final var eventConverter =
        new MongoDbDebeziumEventConverter(cdcMetadataInjector, incrementalOnlyStreamsCatalog, emittedAt, config.getDatabaseConfig());

    final Supplier<AutoCloseableIterator<AirbyteMessage>> incrementalIteratorSupplier = () -> handler.getIncrementalIterators(
        propertiesManager, eventConverter, cdcSavedInfoFetcher, mongoDbCdcStateHandler);
```

Can do something similar as was done in MySQL/MsSQL : 
https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/initialsync/MssqlInitialReadUtil.java#L517

Reply:
thanks for catching this! fixed.